### PR TITLE
fix(datastore-v1): sync pending mutation events with latest synced metadata

### PR DESCRIPTION
**Pull request by 5d**
## Issue \#

issue #1308 
## Description

This is porting from implementation of V2 PR #3377 
## General Checklist

- [x] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Amplify/Categories/DataStore/Model/Internal/Model+Codable.swift [hunk 1]
```diff
@@ -76,6 +76,7 @@ extension Model where Self: Codable {
             resolvedEncoder = encoder
         } else {
             resolvedEncoder = JSONEncoder(dateEncodingStrategy: ModelDateFormatting.encodingStrategy)
+            resolvedEncoder.outputFormatting = .sortedKeys
```
Comment (lawmicha) [inline Amplify/Categories/DataStore/Model/Internal/Model+Codable.swift:79]: This looks like a bug fix to ensure the encoded json always has the same key ordering. It's similar to the changes in  https://github.com/aws-amplify/amplify-swift/pull/3057/files#diff-d149ce8efcd8be113dc6cdd44e9e0142ae3853f49cf99a7390c3c50048fd035c that was merged into `main`. Let's keep the code aligned if we can, ie.
```swift
if isKnownUniquelyReferenced(&resolvedEncoder) {
      resolvedEncoder.outputFormatting = .sortedKeys
  }
```

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift [hunk 2]
```diff
@@ -109,7 +112,7 @@ class SyncMutationToCloudOperation: AsynchronousOperation {
         authType: AWSAuthorizationType? = nil
     ) -> GraphQLRequest<MutationSync<AnyModel>>? {
         var request: GraphQLRequest<MutationSync<AnyModel>>
-
+        let version = getLatestSyncMetadata()?.version
```
Comment (lawmicha) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift:115]: let's keep the code aligned, however small.. from https://github.com/aws-amplify/amplify-swift/pull/3377/files#diff-e22aa4a69639447ae34d5937d1db5813f9d6c878e426206ef3bb85e4b0600973 would be 
```
let latestSyncMetadata = getLatestSyncMetadata()
```

Comment (5d) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift:115]: sure, will update the PR.

File: Amplify/Categories/DataStore/Model/Internal/Model+Codable.swift (hunk 1 above)
Comment (5d) [reply Amplify/Categories/DataStore/Model/Internal/Model+Codable.swift:79]: I will pull the encorder order changes in a seperate PR.

**Review by lawmicha — APPROVED** [review]

**Pull request merged**